Parsers/ASimAuditEvent/Parsers/vimAuditEventCrowdStrikeFalcon.yaml

```diff
@@ -0,0 +1,192 @@
+      "UpdateAPIClient", "Update API Client", "Set", "", "API Client", "Service"
+  ];
+  let EventSeverityLookup = datatable (LogSeverity: string, EventSeverity: string)
+      [
+      "0", "Informational",
+      "1", "Informational",
+      "2", "Low",
+      "3", "Medium",
+      "4", "High",
+      "5", "High"
+  ];
+  let UserAuditActivities = dynamic(["delete_report_execution", "delete_scheduled_report", "update_scheduled_report", "create_scheduled_report", "update_class_action", "update_policy", "enable_policy", "create_policy", "remove_rule_group", "create_rule_group", "delete_rule_group", "add_rule_group", "delete_rule", "update_rule", "create_rule", "disable_policy", "delete_policy", "update_priority", "assign_policy", "remove_policy", "ip_rules_added", "ip_rules_removed"]);
+  let AuthAuditActivities = dynamic(["CreateAPIClient", "UpdateAPIClient"]);
+  let parser = (starttime: datetime=datetime(null), endtime: datetime=datetime(null), srcipaddr_has_any_prefix: dynamic=dynamic([]), eventtype_in: dynamic=dynamic([]), eventresult: string='*', actorusername_has_any: dynamic=dynamic([]), operation_has_any: dynamic=dynamic([]), object_has_any: dynamic=dynamic([]), newvalue_has_any: dynamic=dynamic([]), disabled: bool = false) {
+      let allData = CommonSecurityLog
+          | where not(disabled)
+          | where ((isnull(starttime) or TimeGenerated >= starttime) and (isnull(endtime) or TimeGenerated <= endtime)) 
+          | where (DeviceVendor == "CrowdStrike" and DeviceProduct == "FalconHost")
+          | where (DeviceEventClassID == "UserActivityAuditEvent" and Activity in (UserAuditActivities)) or (DeviceEventCategory == "AuthActivityAuditEvent" and Activity in (AuthAuditActivities))
+          | where array_length(newvalue_has_any) == 0 
+              and (array_length(srcipaddr_has_any_prefix) == 0 or has_any_ipv4_prefix(DestinationTranslatedAddress, srcipaddr_has_any_prefix))
+              and (array_length(actorusername_has_any) == 0 or DestinationUserName has_any (actorusername_has_any))
+              and (array_length(object_has_any) == 0 or Activity has_any (object_has_any))
+          | lookup EventFieldsLookup on Activity
+          | lookup EventSeverityLookup on LogSeverity
+          | extend EventType = EventType_lookup;
+      let useractivitydata = allData
+          | where (DeviceEventClassID == "UserActivityAuditEvent" and Activity in (UserAuditActivities)) 
+          | extend
+              EventStartTime = unixtime_milliseconds_todatetime(tolong(ReceiptTime))
+          | project-rename EventOriginalType = DeviceEventClassID;
+      let authactivitydata = allData
+          | where (DeviceEventCategory == "AuthActivityAuditEvent" and Activity in (AuthAuditActivities))
+          | extend
+              EventStartTime = todatetime(DeviceCustomDate1)
+          | project-rename
+              EventOriginalType  = DeviceEventCategory;
+      union useractivitydata, authactivitydata
```

Comment:
@niralishah-crest 
Please use Union only when it can't be avoided. This impacts the parser performance. Please consider using switch when logic of only few fields are different. Like in this case I see the logic is different for only two fields which could be handled by switch or iff statement for each field.

Reply:
Done